tests/autorestart_test/test_container_autorestart.py

```diff
@@ -0,0 +1,332 @@
+    """
+    @summary: Get container names and its autorestart states by analyzing 
+              the command output of "show container feature autorestart"
+    @return:  A dictionary where keys are the names of containers which have the 
+              autorestart feature implemented and values are the autorestart feature
+              state for that container
+    """
+    container_autorestart_states = {}
+
+    show_cmd_output = duthost.shell(CMD_CONTAINER_FEATURE_AUTORESTART)
+    for line in show_cmd_output["stdout_lines"]:
+        container_name = line.split()[0].strip()
+        container_state = line.split()[1].strip('\n')
+        if container_state in ["enabled", "disabled"]:
+           container_autorestart_states[container_name] = container_state
+
+    return container_autorestart_states
+
+def get_disabled_container_list(duthost):
+    """
+    @summary: Get the container/service names which are disabled by default
+    @return: A list includes the names of disabled containers/services
+    """
+    disabled_containers = []
+
+    show_cmd_output = duthost.shell(CMD_FEATURE)
+    for line in show_cmd_output["stdout_lines"]:
+        container_name = line.split()[0].strip()
+        container_state = line.split()[1].strip('\n')
+        if container_state in ["enabled", "disabled"]:
+            disabled_containers.append(container_name)
+
+    return disabled_containers
+
+def is_container_running(duthost, container_name):
+    """
+    @summary: Decide whether the container is running or not
+    @return:  Boolean value. True represents the container is running
+
+    """
+    is_running = duthost.shell("docker inspect -f \{{\{{.State.Running\}}\}} {}".format(container_name))
+    if is_running["stdout_lines"][0].strip('\n') == "true":
+        return True
+    return False
+
+def get_program_status(duthost, container_name, program_name):
+    """
+    @summary: Return the status of a program in the specified container
+    @return: "RUNNING" or "EXITING" represents the program is in the running
+             or exited status
+    """
+    program_status = None
+
+    process_list = duthost.shell("docker exec {} supervisorctl status".format(container_name))
+    for process_info in process_list["stdout_lines"]:
+        if process_info.find(program_name) != -1:
+            program_status = process_info.split()[1].strip()
+            break
+
+    return program_status
+
+def kill_process_by_pid(duthost, container_name, program_name, program_status, program_pid):
+    """
+    @summary: Kill a process in the specified container by its pid
+
+    """
+    if program_status == "RUNNING":
+        duthost.shell("docker exec {} kill -SIGKILL {}".format(container_name, program_pid)) 
+    elif program_status in ["EXITED", "STOPPED", "STARTING"]:
+        pytest.fail("{} in {} is in the {} state not in the running state" \
+                .format(program_name, container_name, program_status))
```

Comment:
This line should be indented to visually align with the opening `"` in the line above. There are many occurrences of this and other PEP8 violations throughout the file.

Until we have an automated linting solution, I also suggest passing all Python code through http://pep8online.com/ and fixing any PEP8 violations it detects. Note that you can ignore "line too long" warnings; just use your best judgement if a line is over 100 chars.

Reply:
I checked the source code against PEP8 coding standard and fixed the issues such as trailing whitespace, indentation. I will find a solution to configure the editor to avoid such issues.

Reply:
You can look into installing [autopep8](https://pypi.org/project/autopep8/) on your dev machine.

When you run it, you can specify a longer max line length, like `autopep8 --max-line-length 100` or `autopep8 --max-line-length 120`. I think we should not go over 120 chars.